docs/spec/typeddict.rst

```diff
@@ -27,13 +27,13 @@ supported by ``typing.NamedTuple``.  Other features include
 TypedDict inheritance and totality (specifying whether keys are
 required or not).
 
-This section also provides a sketch of how a type checker is expected
-to support type checking operations involving TypedDict objects.
-Similar to :pep:`484`, this discussion is left somewhat vague on purpose,
-to allow experimentation with a wide variety of different type
-checking approaches.  In particular, type compatibility should be
-based on structural compatibility: a more specific TypedDict type can
-be compatible with a smaller (more general) TypedDict type.
+This section also provides a sketch of how a type checker is expected to
+support type checking operations involving TypedDict objects. Similar to
+:pep:`484`, this discussion is left somewhat vague on purpose, to allow
+experimentation with a wide variety of different type checking approaches.  In
```

Comment:
```suggestion
experimentation with a wide variety of different type-checking approaches.  In
```

Reply:
I'm going to consider this out of scope for this PR; I don't want this PR to become a global consistency pass on use of hyphens. That can be a separate PR :)

The "concepts" doc added in this PR consistently doesn't use a hyphen in "type checker" or "type checking" -- I thought that was due to your prior review comments, but I could be remembering wrong.

Reply:
> The "concepts" doc added in this PR consistently doesn't use a hyphen in "type checker" or "type checking" -- I thought that was due to your prior review comments, but I could be remembering wrong.

It probably was due to those prior review comments! Unfortunately it's dependent on context whether a hyphen is appropriate or not. In most cases, "type checking" is used as a noun, in which case it should not be hyphenated. Here, it's not... though you could debate whether it's being used as an adjective or whether it's part of a larger compound noun ("type-checking approaches"). In either case, I think a hyphen helps clarify that "type checking" are intended to bind more tightly than "checking approaches".

But you're right that this is a pre-existing issue, so I'm okay with this being ignored :-)